stages/test/test_gunzip.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/python3
+
+import subprocess
+from unittest import mock
+
+import pytest
+
+from osbuild import testutil
+from osbuild.testutil import has_executable
+
+STAGE_NAME = "org.osbuild.gunzip"
+
+
+# Prepare dataset containing good and bad API call parameters
+@pytest.mark.parametrize("test_data, expected_err", [
+    # Bad API parameters
+    ({}, "'path' is a required property"),
+    ({"path": 123}, "123 is not of type 'string'"),
+    # Good API parameters
+    ({"path": "/tmp/"}, ""),
+    ({"path": "/boot/initramfs-5.14.0-247.el9.x86_64.img"}, ""),
+])
+# This test validates only API calls using correct and incorrect queries
+def test_schema_validation_gunzip(stage_schema, test_data, expected_err):
+    test_input = {
+        "type": STAGE_NAME,
+        "options": {
+        }
+    }
+
+    test_input["options"].update(test_data)
+    res = stage_schema.validate(test_input)
+
+    if expected_err == "":
+        assert res.valid is True, f"err: {[e.as_dict() for e in res.errors]}"
+    else:
+        assert res.valid is False
+        testutil.assert_jsonschema_error_contains(res, expected_err, expected_num_errs=1)
+
+
+# This test creates dummy file, compresses it and then decompresses.
+# End result checksum is compared with initial file checksum.
+@pytest.mark.skipif(not has_executable("gunzip"), reason="gunzip")
+@pytest.mark.skipif(not has_executable("gzip"), reason="gzip")
+def test_gunzip_integration(tmp_path, stage_module):
+    fake_file = "fake_file.txt"
+    fake_archive = f"{fake_file}.gz"
+    fake_input_path = tmp_path / fake_file
+
+    with fake_input_path.open("w") as fp:
```

Comment:
I'm not sure we need the md5sum and the 10mb file here, just decompressing a small file and comparing the content is probably good enough. With that the test can be written in a slightly more compact way maybe? Here is one possible idea:
```python
# This test creates a compressed dummy file then decompresses and
# compares the content.
@pytest.mark.skipif(not has_executable("gunzip"), reason="gunzip")
def test_gunzip_integration(tmp_path, stage_module):
    fake_archive_path = tmp_path / "some-file.gz"
    fake_file_content = b"Some random text written to a file"
    with gzip.GzipFile(fake_archive_path, "w") as fp:
        fp.write(fake_file_content)

    inp = {
        "file": {
            "path": tmp_path,
            "data": {
                "files": {
                    fake_archive_path.name: "Some test file"
                }
            },
        },
    }
    expected_output_path = tmp_path / "tree" / "output.txt"
    expected_output_path.parent.mkdir()
    options = {
        "path": expected_output_path.name
    }
    stage_module.main(inp, expected_output_path.parent, options)
    assert expected_output_path.read_bytes() == fake_file_content

```
(needs a import gzip at the top too)
(full diff: [pr1689.diff.txt](https://github.com/user-attachments/files/16014491/pr1689.diff.txt))

wdyt?

Reply:
Done